tool/src/c2/formatter.rs

```diff
@@ -134,7 +134,7 @@ impl<'tcx> CFormatter<'tcx> {
     /// Resolve and format a type's destructor
     pub fn fmt_dtor_name(&self, ty: TypeId) -> String {
         let ty_name = self.fmt_type_name(ty);
-        format!("{ty_name}_destroy").into()
+        format!("{ty_name}_destroy")
```

Comment:
wait, this is incorrect, it should be using the type's ABI name

Reply:
Alternatively we should persist the dtor name to the AST/HIR CustomType, since we already use ABI names directly pulled from Method.

Reply:
The desctructors ABI name uses the type's ABI name. The name uses the name.

Reply:
Wait, aren't the other backends calling this code?

Reply:
Not necessary for this PR, but in the medium term we ought to be able to decouple backends from the C backend. Might do this as a separate PR.

Reply:
Yes, we probably want an ABI formatter.

Reply:
I think ideally if it lives on the HIR we don't even need a formatter, since it's entirely method/dtor names. The former is already in the AST, the latter is already accessible via a method on the AST.